test/integration/controller/jobset_controller_test.go

```diff
@@ -267,6 +267,20 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("jobset should not succeed if any job is not completed", &testCase{
+			makeJobSet: testJobSet,
+			updates: []*update{
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						ginkgo.By("completing all but 1 job")
+						for i := 0; i < len(jobList.Items)-1; i++ {
+							completeJob(&jobList.Items[i])
+						}
+					},
+					expectedJobSetCondition: "", // active
```

Comment:
The test skips doing the check when the condition is not set. Also, the check currently does "Eventually", in this case it needs to be "Consistently" for few seconds to ensure that the controller indeed didn't attempt to complete the jobset (i.e., the test didn't just check the condition before the controller tried to set it)

Side note: Please ensure that the test fails without the fix to verify that we are properly covering the bug.

Reply:
I addressed this issue and also refactored the way our integration+e2e tests check JobSet statuses to be more flexible and use shared logic.

Also updated e2e bash script to reduce flakiness.

Let me know if you have any thoughts.